readalongs/align.py

```diff
@@ -426,47 +423,50 @@ def align_sequence(
         write_audio_to_file(audio_segment, save_temps + ".wav" + i_suffix)
 
     # Configure soundswallower for this sequence's dict and fsg
-    asr_config.set_string("-dict", dict_file.name)
-    asr_config.set_string("-fsg", fsg_file.name)
+    asr_config["dict"] = dict_file.name
+    asr_config["fsg"] = fsg_file.name
 
     ps = soundswallower.Decoder(asr_config)
     # Align this word sequence
     ps.start_utt()
     ps.process_raw(audio_segment.raw_data, no_search=False, full_utt=True)
     ps.end_utt()
 
-    return ps.seg()
+    return ps.seg
 
 
 def process_segmentation(
-    segmentation: Iterable[soundswallower.Segment],
+    segmentation: Iterable[soundswallower.Seg],
     curr_removed_segments: List[dict],
     noisewords: Set[str],
     frame_size: float,
     debug_aligner: Optional[bool] = False,
 ) -> List[Dict[str, Any]]:
     """Correct output alignments based on do-not-align segments."""
-    aligned_words = []
+    aligned_words: List[Dict[str, Any]] = []
     for word_seg in segmentation:
-        if word_seg.word in noisewords:
+        if word_seg.text in noisewords:
             continue
-        start = word_seg.start_frame * frame_size
-        end = (word_seg.end_frame + 1) * frame_size
-        # change to ms
-        start_ms = int(start * 1000)
-        end_ms = int(end * 1000)
+        start = word_seg.start
```

Comment:
This, and the changes to SoundSwallowerDecoderStub, implies that you're now providing start and duration in seconds (as a float) instead of frames. Definitely a nicer interface. I assume you've made sure the code is all consistent with this change, and I suppose that's unit change that unit testing is exercising extensively (I think).

Reply:
As far as I can tell everything is consistent - that's also why the class name changed from Segment to Seg, to make sure things would fail right away

Reply:
Yeah, that's a good idea. Force it to break when it's semantically different, so old code can't linger around.